### PR TITLE
[ELASTIC] Controller tests

### PR DESCRIPTION
**Pull request by maxhodik**

### CODE REVIEW

File: elastic/src/test/java/com/hodik/elastic/controllers/ProjectControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import java.time.LocalDate;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class ProjectControllerTest {
+    private final Project PROJECT = Project.builder()
```
Comment (Gtm417) [inline elastic/src/test/java/com/hodik/elastic/controllers/ProjectControllerTest.java]: move builder to the private method

File: elastic/src/test/java/com/hodik/elastic/controllers/ProjectControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            .category("Category")
+            .createdDate(LocalDate.of(2020, 7, 5))
+            .description("Description")
+            .finalPlannedDate(LocalDate.of(2025, 12, 31))
+            .isCommercial("Commercial")
+            .isPrivate(false)
+            .startDate(LocalDate.of(2020, 1, 8))
+            .status("Status")
+            .build();
+    private final ProjectDto PROJECT_DTO = ProjectDto.builder()
```
Comment (Gtm417) [inline elastic/src/test/java/com/hodik/elastic/controllers/ProjectControllerTest.java]: same

File: elastic/src/test/java/com/hodik/elastic/controllers/ProjectControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            .id(1L)
+            .name("Name")
+            .category("Category")
+            .createdDate(LocalDate.of(2020, 7, 5))
```
Comment (Gtm417) [inline elastic/src/test/java/com/hodik/elastic/controllers/ProjectControllerTest.java]: move all duplicate numbers, string, dates to constants. Do not duplicate  them

File: elastic/src/test/java/com/hodik/elastic/controllers/ProjectControllerTest.java (hunk 1 above)
Comment (Gtm417) [inline elastic/src/test/java/com/hodik/elastic/controllers/ProjectControllerTest.java]: why upper case if not static? Java code convention

File: elastic/src/test/java/com/hodik/elastic/controllers/ProjectControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+    public static final String IS_COMMERCIAL = "Commercial";
+    public static final boolean IS_PRIVATE = false;
+    public static final String STATUS = "Status";
+
+    private final Project expectedProject = getExpectedProject();
+    private final List<Project> expectedProjectList = List.of(expectedProject);
+
+    private final ProjectDto expectedProjectDto = getProjectDto();
+    private final List<ProjectDto> expectedProjectDtoList = List.of(expectedProjectDto);
+    private final SearchSort searchSort = new SearchSort("Name", true);
```
Comment (Gtm417) [inline elastic/src/test/java/com/hodik/elastic/controllers/ProjectControllerTest.java]: "Name" to constants. Everywhere!

File: elastic/src/test/java/com/hodik/elastic/controllers/ProjectControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+
+    @Mock
+    private ProjectMapper projectMapper;
+
+
+    @InjectMocks
+    private ProjectController projectController;
+
+    @Test
+    void createProject() throws EntityAlreadyExistsException {
```
Comment (Gtm417) [inline elastic/src/test/java/com/hodik/elastic/controllers/ProjectControllerTest.java]: shouldCreateProject()

File: elastic/src/test/java/com/hodik/elastic/controllers/ProjectControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+        //then
+        verify(projectService).createProject(expectedProject);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+    }
+
+    @Test
+    void createProjectException() throws EntityAlreadyExistsException {
+        //given
+        when(projectMapper.convertToProject(expectedProjectDto)).thenReturn(expectedProject);
+        Mockito.doThrow(EntityAlreadyExistsException.class).when(projectService).createProject(expectedProject);
```
Comment (Gtm417) [inline elastic/src/test/java/com/hodik/elastic/controllers/ProjectControllerTest.java:77]: mock findById.isPresent here

File: elastic/src/test/java/com/hodik/elastic/controllers/ProjectControllerTest.java [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+        when(projectMapper.convertToProject(expectedProjectDto)).thenReturn(expectedProject);
+        //when
+        ResponseEntity<HttpStatus> response = projectController.createProject(expectedProjectDto);
+        //then
+        verify(projectService).createProject(expectedProject);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+    }
+
+    @Test
+    void createProjectException() throws EntityAlreadyExistsException {
```
Comment (Gtm417) [inline elastic/src/test/java/com/hodik/elastic/controllers/ProjectControllerTest.java]: shouldThrowExceptionWhenFindByIdIsPresent

File: elastic/src/test/java/com/hodik/elastic/controllers/UserControllerTest.java [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+    public static final String DESCRIPTION = "description";
+    public static final String ROLE_USER = "ROLE_USER";
+    public static final String EMPLOYEE = "EMPLOYEE";
+    public static final String FULL_STACK = "full stack";
+    private final User expectedUser = getUserBuild();
+    private final List<User> expectedUserList = List.of(expectedUser);
+    private final UserDto expectedUserDto = getUserDtoBuild();
+    private final List<UserDto> expectedUserDtoList = List.of(expectedUserDto);
+
+    private final SearchSort searchSort = new SearchSort("Name", true);
```
Comment (Gtm417) [inline elastic/src/test/java/com/hodik/elastic/controllers/UserControllerTest.java]: Name

File: elastic/src/test/java/com/hodik/elastic/controllers/VacancyControllerTest.java [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+    private EsVacancyService vacancyService;
+    @Mock
+    private VacancyMapper vacancyMapper;
+    @InjectMocks
+    private VacancyController vacancyController;
+
+    @Test
+    void createVacancySuccess() throws EntityAlreadyExistsException {
+        //given
+        when(vacancyMapper.convertToVacancy(expectedVacancyDto)).thenReturn(expectedVacancy);
```
Comment (Gtm417) [inline elastic/src/test/java/com/hodik/elastic/controllers/VacancyControllerTest.java:58]: move to setUp(). @BeforeEach

File: elastic/src/test/java/com/hodik/elastic/controllers/ProjectControllerTest.java (hunk 6 above)
Comment (maxhodik) [reply elastic/src/test/java/com/hodik/elastic/controllers/ProjectControllerTest.java:77]: This test for ProjectController. It doesn't have  findById.isPresent method. Probably  the test not need.

File: elastic/src/test/java/com/hodik/elastic/controllers/VacancyControllerTest.java (hunk 9 above)
Comment (maxhodik) [reply elastic/src/test/java/com/hodik/elastic/controllers/VacancyControllerTest.java:58]: no all tests have this string

File: elastic/src/test/java/com/hodik/elastic/controllers/ProjectControllerTest.java (hunk 1 above)
Comment (maxhodik) [reply elastic/src/test/java/com/hodik/elastic/controllers/ProjectControllerTest.java]: done

File: elastic/src/test/java/com/hodik/elastic/controllers/ProjectControllerTest.java (hunk 3 above)
Comment (maxhodik) [reply elastic/src/test/java/com/hodik/elastic/controllers/ProjectControllerTest.java]: done

**Pull request merged**